File: app/src/main/java/com/baronzhang/android/weather/base/BaseView.java
package com.baronzhang.android.weather.base;

/**
 * view interface,所有View(此项目中的View主要是Fragment和自定义的ViewGroup)必须实现此接口
 *
 * @author baronzhang (baron[dot]zhanglei[at]gmail[dot]com)
 */
public interface BaseView<T> {

    void setPresenter(T presenter);
}

File: app/src/main/java/com/baronzhang/android/weather/base/BaseView.kt
package com.baronzhang.android.weather.base

/**
 * view interface,所有View(此项目中的View主要是Fragment和自定义的ViewGroup)必须实现此接口
 *
 * @author baronzhang (微信公众号：BaronTalk)
 */
interface BaseView<T> {

    fun setPresenter(presenter: T)
}
